src/App.tsx

```diff
@@ -17,18 +45,35 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                setQuery={setQuery}
+                query={query}
+                setStatusFilter={setStatusFilter}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoaded && <Loader />}
+              {!isLoaded && (
+                <TodoList
+                  selectedTodoID={selectedTodo}
+                  todos={filteredTodos}
+                  selectedTodo={todo => setSelectedTodo(todo)}
+                  userId={user => setUserId(user)}
+                />
+              )}
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {selectedTodo && (
+        <TodoModal
+          userId={userId}
+          todo={selectedTodo}
+          selectedTodo={setSelectedTodo}
```

Comment:
This callback for closing TodoModal
```suggestion
          onClose={setSelectedTodo}
```